doc/source/whatsnew/v1.1.0.rst

```diff
@@ -36,6 +36,11 @@ For example:
    ser["2014"]
    ser.loc["May 2015"]
 
+Indexing with Nullable Boolean Arrays
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously indexing with a nullable Boolean array would raise a ``ValueError``, however this is now permitted with ``NA`` being treated as ``False``. (:issue:`31503`)
```

Comment:
add an example of the before and the after (style this after similar notes in 1.0.0)